src/Symfony/Component/Lock/Store/MongoDbStore.php

```diff
@@ -0,0 +1,294 @@
+
+        $options = array(
+            'upsert' => true,
+        );
+
+        $key->reduceLifetime($this->initialTtl);
+        try {
+            $this->getCollection()->updateOne($filter, $update, $options);
+        } catch (\MongoDB\Driver\Exception\BulkWriteException $e) {
+            throw new LockConflictedException('Failed to acquire lock', 0, $e);
+        }
+
+        if ($key->isExpired()) {
+            throw new LockExpiredException(sprintf('Failed to store the "%s" lock.', $key));
+        }
+    }
+
+    public function waitAndSave(Key $key)
+    {
+        throw new InvalidArgumentException(sprintf(
+            'The store "%s" does not supports blocking locks.',
+            __CLASS__
+        ));
+    }
+
+    /**
+     * {@inheritdoc}
+     *
+     * db.lock.update(
+     *     {
+     *         _id: "test",
+     *         token: {# unique token #},
+     *         expires_at: {
+     *             $gte : new Date()
+     *         }
+     *     },
+     *     {
+     *         _id: "test",
+     *         expires_at: new Date({# now + ttl #})
+     *     },
+     *     {
+     *         upsert: 1
+     *     }
+     * );
+     */
+    public function putOffExpiration(Key $key, $ttl)
+    {
+        $expiry = $this->createDateTime(microtime(true) + $ttl);
+
+        $filter = array(
+            $this->options['resource_field'] => (string) $key,
+            $this->options['token_field'] => $this->getToken($key),
+            $this->options['expiry_field'] => array(
+                '$gte' => $this->createDateTime(),
+            ),
+        );
+
+        $update = array(
+            '$set' => array(
+                $this->options['resource_field'] => (string) $key,
+                $this->options['expiry_field'] => $expiry,
+            ),
+        );
+
+        $options = array(
+            'upsert' => true,
+        );
+
+        $key->reduceLifetime($ttl);
+        try {
+            $this->getCollection()->updateOne($filter, $update, $options);
+        } catch (\MongoDB\Driver\Exception\BulkWriteException $e) {
+            throw new LockConflictedException('Failed to put off the expiration of the lock', 0, $e);
+        }
+
+        if ($key->isExpired()) {
+            throw new LockExpiredException(sprintf(
+                'Failed to put off the expiration of the "%s" lock within the specified time.',
+                $key
+            ));
+        }
+    }
+
+    /**
+     * {@inheritdoc}
+     *
+     * db.lock.remove({
+     *     _id: "test"
+     * });
+     */
+    public function delete(Key $key)
+    {
+        $filter = array(
+            $this->options['resource_field'] => (string) $key,
+            $this->options['token_field'] => $this->getToken($key),
+        );
+
+        try {
+            $result = $this->getCollection()->deleteOne($filter);
+        } catch (\MongoDB\Driver\Exception\BulkWriteException $e) {
+            throw new LockConflictedException('Failed to delete lock', 0, $e);
```

Comment:
deletion does not throw ConflictException

Reply:
Fixed